lib/vm/src/table.rs

```diff
@@ -79,6 +81,31 @@ pub struct VMTable {
     vm_table_definition: MaybeInstanceOwned<VMTableDefinition>,
 }
 
+impl VMTableProtected
+{
+    /// Get the `VMTableDefinition`.
+    fn get_vm_table_definition(&self) -> NonNull<VMTableDefinition> {
+        self.vm_table_definition.as_ptr()
+    }
+
+    /// Returns the number of allocated elements.
+    pub fn size(&self) -> u32 {
+        unsafe {
+            let td_ptr = self.get_vm_table_definition();
+            let td = td_ptr.as_ref();
+            td.current_elements
+        }
+    }
+}
```

Comment:
Why do we need to protect the table at all?

Reply:
There are race conditions that need to be protected against because the table was converted to a COW (Copy-on-write) pattern. In normal single-threaded code this will be close to a zero cost abstraction but for multi-threaded code it will copy the VMTable before modifying it. For performance reasons I implemented the copying as COW to make it a shallow copy rather than a deep copy as I assume that a large number of the tables do not change between threads

Reply:
the VMTable does not need to be multithread safe anymore so this was removed - instead we follow closer the threads proposal and make a new instance instead